test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -951,14 +951,29 @@ public XContentBuilder shuffleXContent(XContentParser parser, boolean prettyPrin
             xContentBuilder.prettyPrint();
         }
         Token token = parser.currentToken() == null ? parser.nextToken() : parser.currentToken();
-        if (token == Token.START_ARRAY) {
+        if (token.isValue()) {
+            // There is nothing to shuffle so just copy the value:
+            if (token == Token.VALUE_NULL) {
+                xContentBuilder.nullValue();
+            } else if (token == Token.VALUE_BOOLEAN) {
+                xContentBuilder.value(parser.booleanValue());
+            } else if (token == Token.VALUE_NUMBER) {
+                xContentBuilder.value(parser.numberValue());
+            } else if (token == Token.VALUE_STRING) {
+                xContentBuilder.value(parser.text());
+            } else {
+                throw new IllegalArgumentException("unsupported token [" + token + "]");
+            }
+            return xContentBuilder;
```

Comment:
ops, I wonder why we are finding this only now

Reply:
I added this because otherwise `FetchSourceContextTests` test fails. I guess we never shuffled xcontent that had only a top level value (which is valid).

Reply:
I still wonder why this would be a problem. At least as far as I understand this method should only take whole xContent Objects or Arrays, and they should all parse to a map fine (even with just one field). Or are you talking about things like `{ "foo" }`. Is that valid at all? I would like to take a look at the failing test before adding this here, I think this method should rather check that the start token is either Array- or Object-Start and otherwise reject.

Reply:
> Or are you talking about things like { "foo" }. Is that valid at all?

Yes, that is valid. 

I noticed that we didn't have a isolated unit test for `FetchSourceContext`.  This class serializes a bool value directly without adding an enclosing json object. That is because fetch source context is part of a search request under a json field. This causes `FetchSourceContextTests ` to fail, because `shuffleXContent(...)` method returned an empty `XContentBuilder`. This made me change the `shuffleXContent(...)` method. 

I think I can add an assert here to check whether a value is top level and otherwise fail?

Reply:
I took a look at FetchSourceContext and I think its toXContent() method is kind of problematic in a sense that it sometimes renders a complete object (with start/end Object) and sometimes only a value. Given that FetchSourceContext extends ToXContentObject, I think the later is a case that shouldn't happen. Should we move this out of this PR or is FetchSourceContextTests needed? I don't think I would like to special-case the shuffle methods for this.

Reply:
No, this test is not needed for adding mget api to high level rest client. So 👍  to move this test out and its related changes out of this pr.